Breaking change in Apache Thrift NewTSocketFromAddrTimeout

This [commit](https://github.com/apache/thrift/commit/81334cd7345d3b5af165aa875b733a491f1fd5c7) to the Apache Thrift go library adds a socket timeout param to NewTSocketFromAddrTimeout(), which causes a fatal error when compiling **osquery-go** in `transport.go`.

~~~
# github.com/kolide/osquery-go/transport
/go/src/github.com/kolide/osquery-go/transport/transport.go:31:43: not enough arguments in call to thrift.NewTSocketFromAddrTimeout
	have (*net.UnixAddr, time.Duration)
	want (net.Addr, time.Duration, time.Duration)
~~~

I tested on Linux and OSX with go 1.11 and 1.14. Not sure if the Windows version is affected.

The simplest fix looks like using the same timeout value for both the connectTimeout and socketTimeout params, changing [transport.go#L31](https://github.com/kolide/osquery-go/blob/master/transport/transport.go#L31) from:

    trans := thrift.NewTSocketFromAddrTimeout(addr, timeout)

to:

    trans := thrift.NewTSocketFromAddrTimeout(addr, timeout, timeout)

I'm happy to open a PR for this if it looks reasonable.

--- a/go.mod
+++ b/go.mod
@@ -2,10 +2,12 @@ module github.com/kolide/osquery-go
 
 require (
 	github.com/Microsoft/go-winio v0.4.9
-	github.com/apache/thrift v0.12.0
+	github.com/apache/thrift v0.13.1-0.20200603211036-eac4d0c79a5f
 	github.com/davecgh/go-spew v1.1.0 // indirect
 	github.com/pkg/errors v0.8.0
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/stretchr/testify v1.2.2
 	golang.org/x/sys v0.0.0-20180815093151-14742f9018cd // indirect
 )
+
+go 1.13

--- a/server_test.go
+++ b/server_test.go
@@ -124,7 +124,7 @@ func testShutdownDeadlock(t *testing.T) {
 	addr, err := net.ResolveUnixAddr("unix", listenPath)
 	require.Nil(t, err)
 	timeout := 500 * time.Millisecond
-	trans := thrift.NewTSocketFromAddrTimeout(addr, timeout)
+	trans := thrift.NewTSocketFromAddrTimeout(addr, timeout, timeout)
 	err = trans.Open()
 	require.Nil(t, err)
 	client := osquery.NewExtensionManagerClientFactory(trans,

--- a/transport/transport.go
+++ b/transport/transport.go
@@ -28,7 +28,7 @@ func Open(sockPath string, timeout time.Duration) (*thrift.TSocket, error) {
 		return nil, errors.Wrapf(err, "waiting for unix socket to be available: %s", sockPath)
 	}
 
-	trans := thrift.NewTSocketFromAddrTimeout(addr, timeout)
+	trans := thrift.NewTSocketFromAddrTimeout(addr, timeout, timeout)
 	if err := trans.Open(); err != nil {
 		return nil, errors.Wrap(err, "opening socket transport")
 	}

